Modelica/Utilities/Internal.mo

```diff
@@ -240,4 +240,233 @@ especially if the operating system supports Unicode characters.
+    <td>Milli-seconds after seconds</td></tr>
+
+<tr><td>sec</td> <td>0&nbsp;&hellip;&nbsp;59</td>
+    <td>Seconds after minute</td></tr>
+
+<tr><td>min</td> <td>0&nbsp;&hellip;&nbsp;59</td>
+    <td>Minutes after hour</td></tr>
+
+<tr><td>hour</td> <td>0&nbsp;&hellip;&nbsp;23</td>
+    <td>Hours after midnight</td></tr>
+
+<tr><td>day</td> <td>1&nbsp;&hellip;&nbsp;31</td>
+    <td>Day of month</td></tr>
+
+<tr><td>mon</td> <td>1&nbsp;&hellip;&nbsp;12</td>
+    <td>Current month</td></tr>
+
+<tr><td>year</td> <td>&ge; 1970</td>
+    <td>Current year</td></tr>
+</table>
+</blockquote>
+
+<h4>Example</h4>
+<blockquote><pre>
+(ms, sec, min, hour, mon, year) = getTime()   // = (281, 30, 13, 10, 15, 2, 2015)
+                                              // Feb. 15, 2015 at 10:13 after 30.281 s
+</pre></blockquote>
+<h4>Note</h4>
+<p>This function is impure!</p>
+</html>",   revisions="<html>
+<table border=\"1\" cellspacing=\"0\" cellpadding=\"2\">
+<tr><th>Date</th> <th align=\"left\">Description</th></tr>
+
+<tr><td> June 22, 2015 </td>
+    <td>
+
+<table border=\"0\">
+<tr><td>
+         <img src=\"modelica://Modelica/Resources/Images/Logos/dlr_logo.png\">
+</td><td valign=\"bottom\">
+         Initial version implemented by
+         A. Kl&ouml;ckner, F. v.d. Linden, D. Zimmer, M. Otter.<br>
+         <a href=\"http://www.dlr.de/rmc/sr/en\">DLR Institute of System Dynamics and Control</a>
+</td></tr></table>
+</td></tr>
+
+</table>
+</html>"));
+    end getTime;
+
+    pure function localTime "Retrieve the local time (in the local time zone) from elapsed seconds since reference year"
+      extends Modelica.Icons.Function;
+      input Real seconds "Elapsed seconds since epoch_year";
+      input Integer epoch_year = 1970 "Reference year";
+      output Integer ms "Millisecond";
+      output Integer sec "Second";
+      output Integer min "Minute";
+      output Integer hour "Hour";
+      output Integer day "Day";
+      output Integer mon "Month";
+      output Integer year "Year";
+      external "C90" ModelicaTime_localtime(ms, sec, min, hour, day, mon, year, seconds, epoch_year)
+        annotation (IncludeDirectory="modelica://Modelica/Resources/C-Sources", Include="#include \"ModelicaTime.c\"");
+      annotation (Documentation(info="<html>
+TODO
+</html>"));
+    end localTime;
+
+    pure function stringToTime "Retrieve the local time (in the local time zone) from formatted string"
+      extends Modelica.Icons.Function;
+      input String str "Formatted date and time string";
+      input String format = "%Y-%m-%d %H:%M:%S" "Format string passed to strptime";
+      output Integer ms "Millisecond";
+      output Integer sec "Second";
+      output Integer min "Minute";
+      output Integer hour "Hour";
+      output Integer day "Day";
+      output Integer mon "Month";
+      output Integer year "Year";
+      external "C99" ModelicaTime_strptime(ms, sec, min, hour, day, mon, year, str, format)
```

Comment:
I find it confusing that we have C, C90 (which should be the same as C, right? - note that the Modelica specification calls it C89 due to ANSI/ISO difference) and C99. It's not clear why, and since we only have one C-source file where the functions are just defined without anything extra I don't see how it will work.

Reply:
I set "C99" to explicitly enable all format cases of strftime (see http://www.cplusplus.com/reference/ctime/strftime/). I guess we can simply set "C" here, if this is OK for you.

Reply:
Yes, I think "C" would be better - since entirely removing the function for C89/C90 seems excessive. But possibly a comment about the new formats in C99.